src/main/java/de/tum/p2p/api/OnionForwarderAPI.java

```diff
@@ -0,0 +1,102 @@
+package de.tum.p2p.api;
+
+import de.tum.p2p.Peer;
+import de.tum.p2p.api.netty.setup.OnionChannelInitializer;
+import de.tum.p2p.onion.forwarding.OnionForwarder;
+import de.tum.p2p.onion.forwarding.netty.NettyOnionForwarder;
+import de.tum.p2p.onion.forwarding.netty.context.OriginatorContext;
+import de.tum.p2p.rps.InMemoryRandomPeerSampler;
+import de.tum.p2p.rps.RandomPeerSampler;
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+import java.util.List;
+
+import static de.tum.p2p.samples.FakeKeys.fakePublicKey;
+import static de.tum.p2p.util.Nets.localhost;
+import static de.tum.p2p.util.Nets.randUnprivilegedPort;
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * Created by micro on 30/06/2017.
+ */
+
+@Slf4j
+public class OnionForwarderAPI {
```

Comment:
please place this class (and related handlers) in `de.tum.p2p.onion.forwarding.api` package. i believe this will make more sense.